File: 378-kth-smallest-element-in-a-sorted-matrix/kth-smallest-element-in-a-sorted-matrix.py
class Solution:
    def kthSmallest(self, matrix: List[List[int]], k: int) -> int:

        heap = []

        for i in range(len(matrix)):
            for j in range(len(matrix[0])):
                heapq.heappush(heap, matrix[i][j])

        print("print the heap : ", heap)

        for _ in range(k - 1):
            heapq.heappop(heap)

        return heapq.heappop(heap)

File: 378-kth-smallest-element-in-a-sorted-matrix/kth-smallest-element-in-a-sorted-matrix.java
class Solution {
    public int kthSmallest(int[][] matrix, int k) {

        PriorityQueue<Integer> q = new PriorityQueue<>((a, b) -> (b-a));

        for (int i = 0; i < matrix.length; i++) {
            for (int j = 0; j < matrix[0].length; j++) {
                q.add(matrix[i][j]);

                if (q.size() >= k + 1) {
                    q.poll();
                }
            }
        }
        return q.peek();
    }
}